cmd/cstorvolumeclaim/controller.go

```diff
@@ -380,25 +367,21 @@ func (c *CVCController) getCurrentReplicaCount(cvc *apis.CStorVolumeClaim) (int,
 // IsCVRPending look for pending cstorvolume replicas compared to desired
 // replica count. returns true if count doesn't matches.
 func (c *CVCController) IsCVRPending(cvc *apis.CStorVolumeClaim) (bool, error) {
-	rCount := cvc.Annotations["openebs.io/replicaCount"]
-	desiredReplicaCount, err := strconv.Atoi(rCount)
-	if err != nil {
-		return false, err
-	}
+
 	selector := klabels.SelectorFromSet(BaseLabels(cvc))
 	CVRs, err := c.cvrLister.CStorVolumeReplicas(cvc.Namespace).
 		List(selector)
 	if err != nil {
 		return false, merrors.Errorf("failed to list cvr : %v", err)
 	}
 	// TODO: check for greater values
-	return desiredReplicaCount != len(CVRs), nil
+	return cvc.Spec.ReplicaCount != len(CVRs), nil
 }
 
 // BaseLabels returns the base labels we apply to cstorvolumereplicas created
 func BaseLabels(cvc *apis.CStorVolumeClaim) map[string]string {
 	base := map[string]string{
-		pvAnnotaion: cvc.Name,
```

Comment:
you better remove this string definition as well in next PR

Reply:
or in next commit if there are further changes required on this PR

Reply:
this PR seems to be using two different variables for label selector.. better to fix in this PR

Reply:
done